optimizer/src/main/java/com/qubole/quark/planner/parser/ParserFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2015. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.qubole.quark.planner.parser;
+
+import org.apache.calcite.avatica.util.Casing;
+import org.apache.calcite.avatica.util.Quoting;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+
+import com.qubole.quark.QuarkException;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+/**
+ * Created by adeshr on 5/24/16.
+ */
+public class ParserFactory {
+  private SqlQueryParser sqlQueryParser;
+
+  public SqlQueryParser getSqlQueryParser(Properties info, boolean isDirty) throws SQLException {
+    if (isDirty || sqlQueryParser == null) {
+      try {
+        sqlQueryParser = new SqlQueryParser(info);
+      } catch (QuarkException e) {
+        throw new SQLException(e.getMessage(), e);
+      }
+    }
+    return sqlQueryParser;
+  }
+
+  public Parser getParser(String sql, Properties info, boolean isDirty) throws SQLException {
```

Comment:
Looks like factory is caching the SqlQueryParser .... Can you change name 'isDirty' to something that tells user of this function about caching aspect ?